lib/std/math/nextafter.zig

```diff
@@ -0,0 +1,329 @@
+        const sign_bit_mask = @as(Bits, 1) << (@bitSizeOf(T) - 1);
+        const smallest_bits: Bits = @bitCast(std.math.floatTrueMin(T));
+        const y_sign_bit = @as(Bits, @bitCast(y)) & sign_bit_mask;
+        return @bitCast(smallest_bits | y_sign_bit);
+    }
+    if (@typeInfo(T).Float.bits == 80) {
+        // `f80` has an explicitly stored leading "J-bit" between the mantissa and the exponent and
+        // thus requires special handling. This leading bit *must* be set when the value is normal,
+        // an infinity or a NaN and *should* be cleared otherwise.
+
+        const F80Parts = packed struct {
+            mantissa: u64,
+            exponent_and_sign: u16,
+        };
+        const fractional_bits_mask = (1 << std.math.floatFractionalBits(f80)) - 1;
+        const leading_bit_mask = 1 << std.math.floatFractionalBits(f80);
+        const exponent_bits_mask = (1 << std.math.floatExponentBits(f80)) - 1;
+
+        var x_parts: F80Parts = @bitCast(x);
+
+        // Bitwise increment/decrement the mantissa while also taking care to update the exponent if
+        // we overflow the fractional part. This might flip the leading bit; this is intentional.
+        if ((x > 0.0) == (y > x)) {
+            x_parts.mantissa +%= 1;
+            if (x_parts.mantissa & fractional_bits_mask == 0) {
+                x_parts.exponent_and_sign += 1;
+            }
+        } else {
+            if (x_parts.mantissa & fractional_bits_mask == 0) {
+                x_parts.exponent_and_sign -= 1;
+            }
+            x_parts.mantissa -%= 1;
+        }
+
+        // If the new value is normal (indicated by the exponent being all zeroes), the leading bit
+        // might have been cleared from an overflow, so we must ensure that it remains set.
+        if (x_parts.exponent_and_sign & exponent_bits_mask != 0) {
+            x_parts.mantissa |= leading_bit_mask;
+        }
+        // Otherwise, the new value is subnormal and the leading bit will have either flipped from
+        // set to cleared (if the old value was normal) or remained cleared (if the old value was
+        // subnormal), both of which are the outcomes we want.
+
+        return @bitCast(x_parts);
+    } else {
+        var x_bits: Bits = @bitCast(x);
+        if ((x > 0.0) == (y > x)) {
+            x_bits += 1;
+        } else {
+            x_bits -= 1;
+        }
+        return @bitCast(x_bits);
+    }
+}
+
+test "math.nextAfter.int" {
+    try expect(nextAfter(i0, 0, 0) == 0);
+    try expect(nextAfter(u0, 0, 0) == 0);
+    try expect(nextAfter(i1, 0, 0) == 0);
+    try expect(nextAfter(i1, 0, -1) == -1);
+    try expect(nextAfter(i1, -1, -1) == -1);
+    try expect(nextAfter(i1, -1, 0) == 0);
+    try expect(nextAfter(u1, 0, 0) == 0);
+    try expect(nextAfter(u1, 0, 1) == 1);
+    try expect(nextAfter(u1, 1, 1) == 1);
+    try expect(nextAfter(u1, 1, 0) == 0);
+    inline for (.{ i8, i16, i32, i64, i128, i333 }) |T| {
```

Comment:
why not min and max unsigned and signed integers Zig supports?  u/i65535

Reply:
`i/u65535` blows up compile times and memory consumption to a ridiculous level, at least it did for me when I tried using them. I included `i/u333` to have arbitrary bit-width ints represented.